contracts/amm/AmmDataProvider.sol

```diff
@@ -60,42 +60,47 @@ contract AmmDataProvider is IAmmDataProvider {
     /// Given price of bToken we determine what is the largest pool we can create such that
     /// the ratio of its reserves satisfy the given bToken price: Rb / Rw = (1 - Pb) / Pb
     function getVirtualReserves(
-        uint64 seriesId,
+        ISeriesController.Series memory series,
         address ammAddress,
         uint256 collateralTokenBalance,
         uint256 bTokenPrice
     ) public view override returns (uint256, uint256) {
-        // Get residual balances
-        uint256 bTokenBalance = 0; // no bTokens are allowed in the pool
+        //We commenting this out, but in the furute for perpetuals we will need it
+        //We set wTokenBalance = 0
+        //
+        /*
         uint256 wTokenBalance = erc1155Controller.balanceOf(
             ammAddress,
             SeriesLibrary.wTokenIndex(seriesId)
         );
+        if (series.isPutOption) {
+            wTokenBalance = seriesController.getCollateralPerOptionToken(
+                series,
+                wTokenBalance
+            );
+        }*/
 
-        ISeriesController.Series memory series = seriesController.series(
-            seriesId
-        );
+        IMinterAmm amm = IMinterAmm(ammAddress);
+
+        // Get residual balances
+        uint256 bTokenBalance = 0; // no bTokens are allowed in the pool
 
         // For put convert token balances into collateral locked in them
         uint256 lockedUnderlyingValue = 1e18;
         if (series.isPutOption) {
-            wTokenBalance = seriesController.getCollateralPerOptionToken(
-                seriesId,
-                wTokenBalance
-            );
             // TODO: this logic causes the underlying price to be fetched twice from the oracle. Can be optimized.
             lockedUnderlyingValue =
                 (lockedUnderlyingValue * series.strikePrice) /
                 IPriceOracle(addressesProvider.getPriceOracle())
                     .getCurrentPrice(
-                        seriesController.underlyingToken(seriesId),
-                        seriesController.priceToken(seriesId)
+                        address(amm.underlyingToken()),
+                        address(amm.priceToken())
                     );
         }
 
         // Max amount of tokens we can get by adding current balance plus what can be minted from collateral
         uint256 bTokenBalanceMax = bTokenBalance + collateralTokenBalance;
-        uint256 wTokenBalanceMax = wTokenBalance + collateralTokenBalance;
+        uint256 wTokenBalanceMax = 0 + collateralTokenBalance; //We set wTokenBalance = 0
```

Comment:
Do we want to set this to a variable instead of just hard coding it as 0? doesn't really matter but then we have more description of what it is